src/wayland-egldisplay.c

```diff
@@ -815,7 +815,9 @@ static bool getServerProtocolsInfo(struct wl_display *nativeDpy,
     if (ret == 0) {
         wl_display_roundtrip_queue(nativeDpy, queue);
         if (!getDeviceFromDevIdInitialised) {
+#if defined(RTLD_DEFAULT)
```

Comment:
This doesn't seem desirable. I think it's better to fail the build if this isn't defined. Then we can catch issues like the one you're fixing at build time, rather than wonder what's going on at runtime when something seems a little off.

Reply:
I had figured that the #ifdef would let it run on systems that don't have RTLD_DEFAULT, but now that I think about it more, if RTLD_DEFAULT is missing, it's more likely that there's a problem with the meson/configure scripts. In which case, yes, a loud build-time failure would be preferable.